File: Leetcode/1446. Consecutive Characters/solution.c
int maxPower(char * s){
    char * prev = NULL;
    int count = 0;
    int max = 0;

    while(*s != '\0') {
        if(prev && *s == *prev) count++;
        else {
            count = 1;
            prev = s;
        }
        if(count > max) max = count;
        
        s++;
    }
    
    return max;
}

File: Leetcode/1446. Consecutive Characters/solution.py
class Solution:
    def maxPower(self, s: str) -> int:
        prev = None
        counter = 0
        max_count = 0

        for c in s:
            if c == prev:
                counter += 1
            else:
                prev = c
                counter = 1

            max_count = max(counter, max_count)

        return max_count
